invisible_cities/cities/detsim_simulate_electrons.py

```diff
@@ -0,0 +1,52 @@
+import numpy  as np
+
+from typing    import   List, Tuple, Callable
+
+#######################################
+######### ELECTRON SIMULATION #########
+#######################################
+def generate_electrons(energies    : np.array,
+                       wi          : float,
+                       fano_factor : float ) -> np.array:
+    """ generate secondary electrons from energy deposits
+    """
+    nes  = np.array(energies/wi, dtype = int)
+    pois = nes < 10
+    nes[ pois] = np.random.poisson(nes[pois])
+    nes[~pois] = np.round(np.random.normal(nes[~pois], np.sqrt(nes[~pois] * fano_factor)))
+    return nes
+
+
+def drift_electrons(zs             : np.array,
+                    electrons      : np.array,
+                    lifetime       : float,
+                    drift_velocity : float) -> np.array:
+    """ returns number of electrons due to lifetime loses from secondary electrons
+    """
+    ts  = zs / drift_velocity
+    nes = electrons - np.random.poisson(electrons * (1. - np.exp(-ts/lifetime)))
+    nes[nes < 0] = 0
+    return nes
```

Comment:
This function doesn't have the functionality necessary to use a lifetime map as is the standard at the moment. Or am I missing something? I think it needs to be generalised.

Reply:
Thats true, but I think this could be better implemented once the PR #698 is approved, since the way of implementing this callable (that is, a `lifetime(x, y)` function) would be the same as for the psf functions. Once this is done, approved, this function would be easily modified.

Reply:
Including the dependence of lifetime with (x, y) can be done in a future upgrade of the detsim code.